promql/query_logger_test.go

```diff
@@ -136,3 +136,43 @@ func TestMMapFile(t *testing.T) {
 		t.Fatalf("Mmap failed")
 	}
 }
+
+func TestParseBrokenJson(t *testing.T) {
+	for _, tc := range []struct {
+		b []byte
+
+		ok  bool
+		out string
+	}{
+		{
+			b: []byte(""),
+		},
+		{
+			b: []byte("\x00\x00"),
+		},
+		{
+			b: []byte("\x00[\x00"),
+		},
+		{
+			b:   []byte("\x00[]\x00"),
+			ok:  true,
+			out: "[]",
+		},
+		{
+			b:   []byte("[\"up == 0\",\"rate(http_requests[2w]\"]\x00\x00\x00"),
+			ok:  true,
+			out: "[\"up == 0\",\"rate(http_requests[2w]\"]",
+		},
+	} {
+		t.Run("", func(t *testing.T) {
+			ok, out := parseBrokenJson(tc.b)
```

Comment:
Where it makes sense I am advocating for [behaviour tests](https://www.youtube.com/watch?v=EZ05e7EMOLM) using only public API interfaces.

in this case can the test be against `NewActiveQueryTracker` ?

Reply:
I don't think we should stop testing individual function, especially when it'd be more difficult to test as is the case here.